Bio stats assignment.py

```diff
@@ -0,0 +1,143 @@
+"""Bio-Stats 821 Assignment 3 Mohammad Anas."""
+
+
+from os import lseek
+from typing import Union, Any
+from datetime import datetime
+
+
+def parse_data(filename: str) -> dict[str, list[Any]]:
+    """Return data as dictionary, column names as keys, lists as values."""
+    data_lst = []  # a list of  list of each data entry.  O(1)
+    with open(filename, encoding="UTF-8-sig") as file:  # we open the file  O(1)
+        for line in file:  # Loop over each line O(N)
+            line1 = line.split(  # O(N) -> for split function
+                "\t"
+            )  # to split we search for the delimiter through the string and
+            # to do that, we have to loop over it.
+            # we add element to each element.
+            data_lst.append(line1)  # O(1) (amortized)
+
+    # O(1) + O(1) + O(N*(N+1)) complexity uptil now
+
+    data_dict = dict()  # O(1).  we create empty dictionary
+    # we loop over the index of each element in a list (columns on the data)
+
+    for i in range(len(data_lst[1])):  # O(N).
+        values = []  # create a list of values O(1)
+        for row in range(len(data_lst)):  # loop over each row.  O(N)
+            if row == 0:  # check if its a header row O(1)
+                key = data_lst[row][i]  # O(1) # create key variable as a column name
+            else:  # either of the two operation performed
+                values.append(data_lst[row][i])  # O(1)
+
+            # check condition O(1) and performing required operation O(1)
+
+        # append to values if not a column name
+        # our final dictionary will have column names as keys and values will be
+        # a list of all the elements corresponding to a column name for example
+        # Patient ID is column name and a list of all patient IDs is the value of that key
+        data_dict[key] = values  # insert key and value into dictionary
+    # O(1) +O(N((1+N(1+1)) +1))
+    return data_dict  # O(1)
+
+
+# complexity of parse data function
+# O(1) + O(1) + O(N*(N+1)) + O(1) +O(N((1+N(1+1)) +1)) + O(1)
+# O(4) + O(N^2 + N) +  O(2N^2 + 2N)
+# O(3N^2 + 3N + 4) ->  `O(N^2)`
+
+patient = parse_data(  # we load the patients data set
+    "/Users/mohammadanas/Desktop/Duke MIDS/Spring 2021/\
+SoftwareTools/Assignment 2/PatientCorePopulatedTable.txt"
+)
+tests = parse_data(  # we load the test data set
+    "/Users/mohammadanas/Desktop/Duke MIDS/Spring 2021/\
+SoftwareTools/Assignment 2/LabsCorePopulatedTable.txt"
+)
```

Comment:
Don't run these directly in the module, or it will happen when you import this module. You should pass these into the functions below as arguments.